### ISSUE TITLE
about python version

### ISSUE THREAD

**Issue by caogj-0521** [limhoyeon/ToothGroupNetwork#9]
We can not find python version in the project .
Can you tell me what is the version you used, thans!

**Comment by limhoyeon** [comment]
maybe this issue is related to https://github.com/limhoyeon/ToothGroupNetwork/issues/4.
there are issues with CUDA functions not executing properly in some PC environments. I plan to remove the CUDA functions and replace them with Python code. Please bear with me a little longer, or alternatively, you can try reinstalling the 'pointops' library from https://github.com/POSTECH-CVLab/point-transformer/tree/master/lib/pointops.

**Comment by supgy** [comment]
> Hello:
> 
> I'm using the main branch an follow the procedure:
> 
> 1. use challenger dataset to process,
>    2.use processing data to train
>    3.train command:
>    python start_train.py --model_name tgnet_fps --config_path train_configs/tgnet_fps.py --experiment_name tgnet --input_data_dir_path processDir --train_data_split_txt_path split/base_name_train_fold.txt --val_data_split_txt_path split/base_name_val_fold.txt
>    However, the following error occure
>    ![image](https://private-user-images.githubusercontent.com/75820646/254529586-4b403ed0-a10a-4df4-a71b-a18dfa52244d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY4NzI0NzAsIm5iZiI6MTcwNjg3MjE3MCwicGF0aCI6Ii83NTgyMDY0Ni8yNTQ1Mjk1ODYtNGI0MDNlZDAtYTEwYS00ZGY0LWE3MWItYTE4ZGZhNTIyNDRkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjAyVDExMDkzMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFlZTg0OGNhMjRhYTUzMGI3YWVhOWNmZWY5NDEwYWI3YzExMjkyZDU0NTM5ZTk5NmVmNGU1YjAwYTE0Yzk4OTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.HQIHjukRlF6_g9_vEUq4yl5ebIX4TisFxspOGWDHQzM)
> 
> When I comment out the code ‘raise’： the following error occurre ![39d08ca81aeab5de92173beaf461f0a](https://private-user-images.githubusercontent.com/75820646/254746599-48fe30ee-04db-46e8-997a-4e062bd63bbb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY4NzI0NzAsIm5iZiI6MTcwNjg3MjE3MCwicGF0aCI6Ii83NTgyMDY0Ni8yNTQ3NDY1OTktNDhmZTMwZWUtMDRkYi00NmU4LTk5N2EtNGUwNjJiZDYzYmJiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjAyVDExMDkzMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFlMmUyZmE2MGU4YjBlZDNiZTdkODUzYjVkZGJhMmE1MzI5MDk0ZDk5MzdhZGViNTMxNmViNWE2ZDU5ZTY2Y2MmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.JrV5Rjm7sjI_m0b10XYCQsPmyv-ySjGWCVScz0Cwe_w)
> 
> My running environment： The graphics card is RTX 3090 Python 3.7.16 torch: 1.13.1+cu117, cuda 11.7,cudnn 8500 Ubuntu 20.4
> 
> Now I don't know where the problem is, Can you help me?

Hello, have you solved this problem? I encountered the same problem as you. 
My running environment：
The graphics card is RTX 4090
Python 3.6.13
pytorch1.10 cuda11.7 cudnn
Ubuntu 20.4
Hope your reply. :)

**Comment by limhoyeon** [comment]
Did you tried to replace the point ops with "https://github.com/POSTECH-CVLab/point-transformer/tree/master/lib/pointops"??

**Comment by FengZhongLiuDong** [comment]
Hello, have you solved this problem? I encountered the same problem as you.
My running environment：
The graphics card is RTX 4060
Python 3.8
pytorch2.1 cuda11.8 cudnn
Windows

> Hello:
> 
> I'm using the main branch an follow the procedure:
> 
> 1. use challenger dataset to process,
>    2.use processing data to train
>    3.train command:
>    python start_train.py --model_name tgnet_fps --config_path train_configs/tgnet_fps.py --experiment_name tgnet --input_data_dir_path processDir --train_data_split_txt_path split/base_name_train_fold.txt --val_data_split_txt_path split/base_name_val_fold.txt
>    However, the following error occure
>    ![image](https://private-user-images.githubusercontent.com/75820646/254529586-4b403ed0-a10a-4df4-a71b-a18dfa52244d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzIzNTE0NTksIm5iZiI6MTczMjM1MTE1OSwicGF0aCI6Ii83NTgyMDY0Ni8yNTQ1Mjk1ODYtNGI0MDNlZDAtYTEwYS00ZGY0LWE3MWItYTE4ZGZhNTIyNDRkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTIzVDA4MzkxOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY3OTQwYzVlNjYzOWMwMzQ3YmY2MzhjMzc3MGRiM2UxOGQ2YTI3OGZlMTBiOWMzYWE3NjU0Y2FjZmRmNWM3YjgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.lcjeRSbn2wOPX6VOm1L85eu42FlFdmvgwrqu5mHRILM)
> 
> When I comment out the code ‘raise’： the following error occurre ![39d08ca81aeab5de92173beaf461f0a](https://private-user-images.githubusercontent.com/75820646/254746599-48fe30ee-04db-46e8-997a-4e062bd63bbb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzIzNTE0NTksIm5iZiI6MTczMjM1MTE1OSwicGF0aCI6Ii83NTgyMDY0Ni8yNTQ3NDY1OTktNDhmZTMwZWUtMDRkYi00NmU4LTk5N2EtNGUwNjJiZDYzYmJiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTIzVDA4MzkxOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThlOWU1MjkxYzAzOGNiZDA2NjAyZTcxYTA2NDkzNGRlZTc2MWY0YTMyMmIyM2U1MTZiNGUzMDljM2MxYjViYWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.RRbWQy1v9v4UAjurtjqheaZz3AZlCE-JAjVdvekBpLQ)
> 
> My running environment： The graphics card is RTX 3090 Python 3.7.16 torch: 1.13.1+cu117, cuda 11.7,cudnn 8500 Ubuntu 20.4
> 
> Now I don't know where the problem is, Can you help me?

**Comment by ptpam** [comment]
In my case, the issue arises from this [line](https://github.com/limhoyeon/ToothGroupNetwork/blob/d7e16a4b8fca811975c3a4c7c1429bb0a51b3083/preprocess_data.py#L40). I did not explicitly label the jaws as either upper or lower. Consequently, if I mistakenly name the upper jaw as "lower" extension and pass it to this function, it results in an error while training.

**Comment by FengZhongLiuDong** [comment]
@limhoyeon #

> In my case, the issue arises from this [line](https://github.com/limhoyeon/ToothGroupNetwork/blob/d7e16a4b8fca811975c3a4c7c1429bb0a51b3083/preprocess_data.py#L40). I did not explicitly label the jaws as either upper or lower. Consequently, if I mistakenly name the upper jaw as "lower" extension and pass it to this function, it results in an error while training.

Are you suggesting that the upper and lower jaw data need to be processed separately? Could you please provide a specific solution for this? We deeply appreciate your explanation and hope you can assist us further.

**Comment by FengZhongLiuDong** [comment]
Thank you very much for your reply. I must say that this is an excellent piece of work, but the issue still persists during training. First, I can rule out any environment-related problems, as I rented a server on a cloud computing platform with the same configuration as yours, but the same issue still occurs. Then, I tried recompiling PointOps (using the official code), but unfortunately, that didn't solve the problem either. I believe the best way to solve this issue is to reproduce the error. If you have some free time, I would appreciate it if you could run the code from the GitHub repository again. I am also in the process of locating and fixing the error step by step and would appreciate your assistance.

> 就我而言，问题就来自这一[行](https://github.com/limhoyeon/ToothGroupNetwork/blob/d7e16a4b8fca811975c3a4c7c1429bb0a51b3083/preprocess_data.py#L40)。我没有明确地将下颌标记为上颌或下颌。因此，如果我错误地将上颌命名为 “下” 伸展并将其传递给这个函数，则会导致训练时出错。

**Comment by FengZhongLiuDong** [comment]
I made a very simple mistake. When the author reminded me to use the official PointOps library, I got lazy and kept the data processed with the previous version of PointOps, directly using it for training. However, when I debugged and pinpointed the label issue, I suspected that there might have been a problem with the data processing. When reviewing the problematic module of the project again, I discovered issue #7 and confirmed my suspicion (https://github.com/limhoyeon/ToothGroupNetwork/issues/7). The data processed with the current version of the PointOps library only contains one label, which caused the neighborhood index extraction to fail during training.
This is an outstanding piece of work, and I will study it carefully and thoroughly. Once again, I sincerely appreciate the author's help and am deeply grateful
